components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -680,11 +679,11 @@ impl Interpolate for BoxShadow {
     }
 }
 
-impl Interpolate for LengthOrNone {
+impl<T: Interpolate, U> Interpolate for Either<T, U> {
```

Comment:
I think this is wrong, what if we have `Either<Length, Number>`? We'd only be interpolating numbers with this code.